jhove-modules/wave-hul/src/main/java/edu/harvard/hul/ois/jhove/module/WaveModule.java

```diff
@@ -254,7 +252,7 @@ public WaveModule() {
 
 		doc = new Document("Specification of the Broadcast Wave Format (BWF)",
 				DocumentType.REPORT);
-		doc.setIdentifier(new Identifier("EBU Technical Specification 3285",
+		doc.setIdentifier(new Identifier(FORMAT[2],
```

Comment:
Thanks, @david-russo , this is very clear and helpful. As a learning exercise, if I may ask (just to be sure I fully understand) - the idea would be to leave the array as-is to keep a human-readable list of formats but add additional `static final` variables to eliminate the repeated string literals? And by extension, while `FORMAT[0]` is less likely to be problematic, it could be, and in any case if only for stylistic consistency should probably also have a separate `static final`?

Reply:
Spot on. So using the above code as an example, I would create a new variable and place it in the code wherever that string is used with the same meaning (in the case of the tech. spec. name, it's all just for human-readable documentation):

```java
private static final EBU_TECH_SPEC_3285 = "EBU Technical Specification 3285";
...
private static final String[] FORMATS = { "WAVE", ..., EBU_TECH_SPEC_3285, ...
...
doc.setIdentifier(new Identifier(EBU_TECH_SPEC_3285, ...
```

And yes, the fix for `FORMAT[0]` usage is probably just replacing it with references to appropriate `static final` variables.